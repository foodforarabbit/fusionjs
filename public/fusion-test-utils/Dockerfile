FROM uber/web-base-image:1.0.0

WORKDIR /fusion-test-utils

COPY . .

RUN yarn

RUN yarn build-test
