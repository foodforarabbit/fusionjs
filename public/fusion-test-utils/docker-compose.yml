version: '2'
services:
  fusion-test-utils:
    build: .
    volumes:
      - .:/fusion-test-utils
      - /fusion-test-utils/node_modules/
      - /fusion-test-utils/dist/
      - /fusion-test-utils/dist-tests/
