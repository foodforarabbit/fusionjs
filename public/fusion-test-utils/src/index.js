import {mockContext, renderContext} from './mock-context.js';
import simulate from './simulate';
import assert from 'assert';

export function request(app, url, options = {}) {
  if (__BROWSER__) {
    throw new Error(
      '[fusion-test-utils] Request api not support from the browser. Please use `render` instead'
    );
  }
  const ctx = mockContext(url, options);
  return simulate(app, ctx);
}

export function render(app, url, options = {}) {
  const ctx = renderContext(url, options);
  return simulate(app, ctx);
}

// Export test runner functions from jest
let mockFunction, test;
if (typeof it !== 'undefined') {
  // Surface snapshot testing
  assert.matchSnapshot = tree => expect(tree).toMatchSnapshot();

  /* eslint-env node, jest */
  test = (description, callback, ...rest) =>
    it(description, () => callback(assert), ...rest);
  mockFunction = (...args) => jest.fn(...args);
} else {
  const notSupported = () => {
    throw new Error('Canâ€™t import test() when not using the test-app target.');
  };
  test = notSupported;
  mockFunction = notSupported;
}
export {mockFunction, test};
