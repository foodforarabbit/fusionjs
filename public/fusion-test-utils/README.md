# fusion-test-utils

Provides test utility functions for FusionJS

---

```sh
yarn add fusion-test-utils
```

### Example

```js
import App from 'fusion-core';
import {mockContext} from 'fusion-test-utils';

const app = new App();
let ctx = mockContext();
await app.simulate(ctx);
// do assertions on ctx

let ctx = mockContext({path: '/'}); // argument is merged into mock req object
await app.simulate(ctx);
// do assertions on ctx
```


#### Browser context

Browsers send an `accept: 'text/html'` header, which can determine whether a SSR happens. Using the `browser()` method will ensure this header is set.

```js
const app = new App();
let ctx = mockContext.browser();
await app.simulate(ctx);
// do assertions on ctx
```

---
