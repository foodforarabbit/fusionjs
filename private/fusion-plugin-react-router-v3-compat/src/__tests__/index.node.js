import App from 'fusion-core';
import Router from 'fusion-plugin-react-router';
import {test, getSimulator} from 'fusion-test-utils';
import TestRenderer from 'react-test-renderer';

import Plugin from '../plugin.js';

import RoutesFixtures from '../__fixtures__/routes';

test('Normal route config', async assert => {
  const app = new App(RoutesFixtures.normal, el => {
    const tree = TestRenderer.create(el).toJSON();
    assert.matchSnapshot(tree);
  });

  app.register(Plugin);
  app.register(Router);

  const simulator = getSimulator(app);
  simulator.render('/');
  simulator.render('/fancy');
  simulator.render('/stuff');
});

test('Redirection', async assert => {
  const app = new App(RoutesFixtures.redirect, el => {
    TestRenderer.create(el).toJSON();
  });

  app.register(Plugin);
  app.register(Router);

  const simulator = getSimulator(app);
  const ctx = await simulator.render('/nowhere');
  assert.strictEqual(
    ctx.response.headers.location,
    '/somewhere',
    'redirection matches'
  );
});
