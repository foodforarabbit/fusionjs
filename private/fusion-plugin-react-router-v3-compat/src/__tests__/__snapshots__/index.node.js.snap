// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normal route config 1`] = `
<div>
  <div>
    <div>
      Main
    </div>
  </div>
</div>
`;

exports[`Normal route config 2`] = `
<div>
  <div>
    <div>
      Fancy
    </div>
  </div>
</div>
`;

exports[`Normal route config 3`] = `
<div>
  <div>
    <div>
      Stuff
    </div>
  </div>
</div>
`;

exports[`Normal route config 4`] = `
<div>
  <div>
    <div>
      browserHistoryCompat.push: 
      function () { [native code] }
    </div>
  </div>
</div>
`;
